casepro/statistics/signals.py

```diff
@@ -78,3 +78,35 @@ def record_incoming_labelling(sender, instance, action, reverse, model, pk_set,
     elif action == 'pre_clear':
         for label in instance.labels.all():
             DailyCount.record_removal(day, DailyCount.TYPE_INCOMING, label)
+
+
+@receiver(post_save, sender=CaseAction)
+def record_new_case_action(sender, instance, created, **kwargs):
+    org = instance.case.org
+    user = instance.created_by
+    partner = instance.case.assignee
+    case = instance.case
+
+    day = datetime_to_date(instance.created_on, instance.case.org)
+
+    if instance.action == CaseAction.CLOSE:
+        # count the time to close on an org level
+        td = instance.created_on - case.opened_on
+        minutes_since_open = ceil(td.total_seconds() / 60)
+        DailyMinuteTotalCount.record_item(day, minutes_since_open,
+                                          DailyMinuteTotalCount.TYPE_TILL_CLOSED, org)
+
+        if case.assignee == partner:
+            # count the time since case was last assigned to this partner till it was closed
+            if user.partners.filter(id=partner.id).exists():
+                # count the time since this case was (re)assigned to this partner
+                try:
+                    action = case.actions.filter(assignee=partner).latest('created_on')
```

Comment:
Maybe we should also be filtering on the `action` field, which is the type of action, since in another PR we added that the `assignee` field be present on case close actions (and it could be added to other case action types in the future).